lib/ffmpeg/transcoder.rb

```diff
@@ -18,11 +18,22 @@ def self.timeout
       @@timeout
     end
 
+    def requires_pre_encode
+      # Stitches and trims with dynamic resolution require pre-encoding
+      @movie.requires_pre_encode = @movie.paths.size > 1 || (@movie.has_dynamic_resolution && @transcoder_options[:permit_dynamic_resolution_pre_encode])
```

Comment:
nit: Should this be updated each time it's called, or should it only be updated if it hasn't been set?
```suggestion
      @movie.requires_pre_encode ||= @movie.paths.size > 1 || (@movie.has_dynamic_resolution && @transcoder_options[:permit_dynamic_resolution_pre_encode])
```
If we did go that route, you'd want to update this
```ruby
     @requires_pre_encode = false
```
To this
```ruby     
     @requires_pre_encode = nil
```